host_core/lib/host_core/providers/provider_supervisor.ex

```diff
@@ -247,12 +247,11 @@ defmodule HostCore.Providers.ProviderSupervisor do
     }
 
     with {:ok, {pid, _}} <- VirtualHost.lookup(host_id),
-         config <- VirtualHost.config(pid),
-         labels <- VirtualHost.labels(pid),
+         full_state <- VirtualHost.full_state(pid),
```

Comment:
From what I can gather, the host never calls `start_executable_provider` in a blocking context (only in a `ControlInterfaceTaskSupervisor` task) so this shouldn't block